### PR TITLE
HHH-17047 - Follow up tasks for Gradle 8.2 upgrade

### PR DESCRIPTION
**Pull request by marko-bekhta**
- tweak how see additional links section is rendered
- update external javadoc links

### CODE REVIEW

File: local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java [hunk 1]
```diff
@@ -216,13 +216,13 @@ private static Elements cleanupFieldJavadocElement(Element fieldJavadocElement,
...
 			usefulDocsPart.add( actualJavadocs );
 		}
-		cleanupSeeLinks( fieldJavadocElement.selectFirst( "ul.see-list" ) )
-				.ifPresent( ul -> {
-					// add a see also
-					usefulDocsPart.add( new Element( "div" ).appendChild( new Element( "b" ).text( "See also:" ) ) );
-					// add the list of links:
-					usefulDocsPart.add( ul );
-				} );
+		usefulDocsPart.add( new Element( "div" )
```
Comment (sebersole) [inline local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java:219]: If we are building that directly into the "javadoc" text, maybe we should remove the corresponding (and no longer used) attributes from `SettingDescriptor`?

Comment (marko-bekhta) [reply local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java:219]: It was tempting 😃, but I thought you were planning to use them elsewhere... but if not -- sure 😃

File: documentation/documentation.gradle [hunk 2]
```diff
@@ -225,9 +225,10 @@ def aggregateJavadocsTask = tasks.register( "aggregateJavadocs", Javadoc ) {
 
 		links = [
 				'https://docs.oracle.com/en/java/javase/11/docs/api/',
-				'https://docs.jboss.org/hibernate/beanvalidation/spec/2.0/api/',
-				'https://docs.jboss.org/cdi/api/2.0/',
-				'https://jakarta.ee/specifications/platform/8/apidocs/'
+				'https://jakarta.ee/specifications/bean-validation/3.0/apidocs/',
```
Comment (sebersole) [inline documentation/documentation.gradle:228]: Good catch!

File: local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java (hunk 1 above)
Comment (sebersole) [reply local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java:216]: The constant class and field name are possibly useful.  But the Javadoc link attribute is useless now

Comment (marko-bekhta) [reply local-build-plugins/src/main/java/org/hibernate/orm/properties/SettingsCollector.java:218]: ok, the link is gone 😃

**Pull request merged**